cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -352,12 +362,20 @@ private LocalFile uploadFile(String bucket, String targetFilePath,
       }
       // Precondition fails means the blob already exists, most likely happens due to retries
       // https://cloud.google.com/storage/docs/request-preconditions#special-case
-      // Overwrite the file
+      // Overwrite the file if isCacheable is set to true
       Blob blob = storage.get(blobId);
-      BlobInfo existingBlobInfo = BlobInfo.newBuilder(blob.getBlobId()).setContentType(contentType).build();
-      uploadToGCS(localFile.getURI(), storage, existingBlobInfo, Storage.BlobWriteOption.generationMatch());
-      LOG.debug("Successfully uploaded file {} to gs://{}/{} by overwriting due to conflict",
-                localFile.getURI(), bucket, targetFilePath);
+      BlobInfo.Builder existingBlobInfoBuilder = BlobInfo.newBuilder(blob.getBlobId()).setContentType(contentType);
+      if (!isCacheable) {
```

Comment:
do you think it would be useful to also add a hidden argument to turn this on/off in the dataproc provisioner? May be easier to do experiments, as you could have two profiles, one with it on and one with it off and wouldn't need to restart cdap.

Reply:
Great idea. I'll address it in a separate PR.